packages/api/src/store/rabbit-queue.test.ts

```diff
@@ -9,6 +10,20 @@ async function sleep(duration) {
   });
 }
 
+function semaphore() {
```

Comment:
Neat. Want to put this in a shared helper someplace where we could import it in other tests?

Reply:
Sure! I'll even move/export the `sleep` one too and avoid repeating it on stream tests